types/nullable_test.go

```diff
@@ -0,0 +1,248 @@
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, false, obj.SimpleInt.Value.ReplicaCount.IsSet(), "replica count should not be set")
+				assert.Equalf(t, false, obj.SimpleInt.Value.ReplicaCount.IsNull(), "replica count should not be null")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsSet(), "name should not be set")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsNull(), "name should not be null")
+				assert.Equalf(t, false, obj.StringList.IsSet(), "string list should not be set")
+				assert.Equalf(t, false, obj.StringList.IsNull(), "string list should not be null")
+			},
+		},
+
+		{
+			name:      "replica count having non null value",
+			jsonInput: []byte(`{"simple_int":{"replicaCount":1}}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, false, obj.SimpleInt.Value.ReplicaCount.IsNull(), "replica count should NOT be null")
+				assert.Equalf(t, true, obj.SimpleInt.Value.ReplicaCount.IsSet(), "replica count should be set")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsSet(), "name should NOT be set")
+				assert.Equalf(t, false, obj.SimpleString.Value.Name.IsNull(), "name should NOT be null")
+				gotValue, isNull := obj.SimpleInt.Value.ReplicaCount.Get()
+				assert.Equalf(t, false, isNull, "replica count should NOT be null")
+				assert.Equalf(t, 1, gotValue, "replica count should be 1")
+			},
+		},
+
+		{
+			name:      "string list having null value",
+			jsonInput: []byte(`{"string_list": null}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, true, obj.StringList.IsSet(), "string_list should be set")
+				assert.Equalf(t, true, obj.StringList.IsNull(), "string_list should be null")
+			},
+		},
+
+		{
+			name:      "string list having non null value",
+			jsonInput: []byte(`{"string_list": ["foo", "bar"]}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, true, obj.StringList.IsSet(), "string_list should be set")
+				assert.Equalf(t, false, obj.StringList.IsNull(), "string_list should not be null")
+				gotStringList, isNull := obj.StringList.Get()
+				assert.Equalf(t, false, isNull, "string_list should not be null")
+				assert.Equalf(t, []string{"foo", "bar"}, gotStringList, "string_list should have the values as provided in the jSON")
+
+			},
+		},
+
+		{
+			name:      "set string list having empty value",
+			jsonInput: []byte(`{"string_list":[]}`),
+			assert: func(obj TestComplex, t *testing.T) {
+				assert.Equalf(t, true, obj.StringList.IsSet(), "string_list should be set")
+				assert.Equalf(t, false, obj.StringList.IsNull(), "string_list should not be null")
+				gotStringList, isNull := obj.StringList.Get()
+				assert.Equalf(t, false, isNull, "string_list should not be null")
+				assert.Equalf(t, []string{}, gotStringList, "string_list should have the values as provided in the jSON")
+
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t1 *testing.T) {
```

Comment:
```suggestion
		t.Run(tt.name, func(t *testing.T) {
```

Otherwise we're not passing in the correct type.

Shadowing here is fine (in my opinion - happy to chat if there are things to be aware of)

Reply:
typo.. thanks. 
Will fix that.